x-pack/plugins/uptime/public/hooks/use_composite_image.ts

```diff
@@ -31,25 +62,34 @@ export const useCompositeImage = (
   onComposeImageSuccess: React.Dispatch<string | undefined>,
   imageData?: string
 ): void => {
+  const dispatch = useDispatch();
+  const { blocks }: { blocks: ScreenshotBlockCache } = useSelector(syntheticsSelector);
+
+  React.useEffect(() => {
+    dispatch(
+      fetchBlocksAction(imgRef.ref.screenshotRef.screenshot_ref.blocks.map(({ hash }) => hash))
```

Comment:
I guess TypeScript has made me cocky 😆. You do have a good point here though; I think the type we're using here has potential for simplification. I think the `ref` (or perhaps `screenshot_ref`?) field might be able to be refactored away. I haven't done it now because the type is used in many places and have been wary of revising it; now that we're toward the end stages of the project I could modify it. I also have not been pleased with how much nesting there is, but like you say, with TS we can have more confidence.